File: framework/lib/solid_pen.dart
part of uxml;

/**
 * This class represents a pen with one solid color.
 *
 * @author sanjayc@ (Sanjay Chouksey)
 * @author ferhat@ (Ferhat Buyukkokten)
 */
class SolidPen extends Pen {

  /** The color of the brush. */
  Color _color;

  /**
   * Constructs a solid color brush.
   */
  SolidPen(Color color, num thickness) : super(thickness) {
    _color = (color == null ? Color.BLACK : color);
  }

  /**
   * Gets or sets the color value.
   */
  Color get color {
    return _color;
  }

  set color(Color value) {
    if (value == null) {
      throw new ArgumentError("Invalid color value");
    }
    _color = value;
  }

  /**
   * Creates a new pen that is the clone of this pen.
   */
  Pen clone() {
    return new SolidPen(color, _thickness);
  }
}

File: plugin/com.hello.uxml.tools.ui/src/com/hello/uxml/tools/framework/graphics/SolidPen.java
package com.hello.uxml.tools.framework.graphics;

/**
 * Represents attributes of a line drawing.
 *
 * @author ferhat
 */
public class SolidPen extends Pen {
  /** Color of pen */
  private Color color;

  /**
   * Constructs a black pen of 1.0 thickness.
   */
  public SolidPen() {
    color = Color.fromRGB(0);
  }

  /**
   * Constructor
   */
  public SolidPen(Color color, double thickness) {
    this.color = color;
    this.thickness = thickness;
  }

  /**
   *  Sets/returns color of pen
   */
  public void setColor(Color value) {
    color = value;
  }

  public Color getColor() {
    return color;
  }
}
